metals/src/main/scala/scala/meta/internal/implementation/SuperMethodProvider.scala

```diff
@@ -0,0 +1,276 @@
+  ): List[SymbolWithAsSeenFrom] = {
+    if (skipSymbols.contains(symbolInformation.symbol)) {
+      List.empty
+    } else {
+      skipSymbols += symbolInformation.symbol
+      symbolInformation.signature match {
+        case classSignature: ClassSignature =>
+          val parents = classSignature.parents
+            .collect { case x: TypeRef => x }
+            .filterNot(typeRef =>
+              stopSymbols.contains(typeRef.symbol) || skipSymbols
+                .contains(typeRef.symbol)
+            )
+          val parentsHierarchy = parents
+            .flatMap(p => findSymbol(p.symbol).map((_, p)))
+            .map {
+              case (si, p) =>
+                val parentASF = AsSeenFrom.calculateAsSeenFrom(
+                  p,
+                  classSignature.typeParameters
+                )
+                getSuperClasses(si, findSymbol, skipSymbols, parentASF)
+            }
+            .toList
+            .reverse
+            .flatten
+          SymbolWithAsSeenFrom(symbolInformation, asSeenFrom) +: parentsHierarchy
+        case sig: TypeSignature =>
+          val upperBound = sig.upperBound.asInstanceOf[TypeRef]
+          findSymbol(upperBound.symbol)
+            .filterNot(s => skipSymbols.contains(s.symbol))
+            .map(si => {
+              val parentASF =
+                AsSeenFrom.calculateAsSeenFrom(upperBound, sig.typeParameters)
+              getSuperClasses(si, findSymbol, skipSymbols, parentASF)
+            })
+            .getOrElse(List.empty)
+        case _ =>
+          List.empty
+      }
+    }
+  }
+
+  private def calculateClassSuperHierarchyWithCache(
+      classSymbolInformation: SymbolInformation,
+      cache: LensGoSuperCache,
+      findSymbol: String => Option[SymbolInformation]
+  ): List[SymbolWithAsSeenFrom] = {
+    cache.get(classSymbolInformation.symbol) match {
+      case Some(value) => value
+      case None =>
+        val value =
+          calculateClassSuperHierarchy(classSymbolInformation, findSymbol)
+        cache(classSymbolInformation.symbol) = value
+        value
+    }
+
+  }
+
+  private def calculateClassSuperHierarchy(
+      classSymbolInformation: SymbolInformation,
+      findSymbol: String => Option[SymbolInformation]
+  ): List[SymbolWithAsSeenFrom] = {
+    getSuperClasses(
+      classSymbolInformation,
+      findSymbol,
+      m.Set[String](),
+      Map()
+    )
+  }
+
+  private def findSuperForMethodOrFieldChecked(
+      msi: SymbolInformation,
+      documentWithPath: TextDocumentWithPath,
+      cache: LensGoSuperCache,
+      findSymbol: String => Option[SymbolInformation]
+  ): Option[String] = {
+    val classSymbolInformation =
+      findClassInfo(
+        msi.symbol,
+        msi.symbol.owner,
+        documentWithPath.textDocument,
+        findSymbol
+      )
+    val methodInfo = msi.signature.asInstanceOf[MethodSignature]
+
+    val result = for {
+      si <- classSymbolInformation.toIterable
```

Comment:
Can this just be `symbolInformation` rather than `si`

Reply:
I changed `classSymbolInformation` -> `classSymbolInformationOption`
and `si` -> `classSymbolInformation`
what you think?

Reply:
yea, I think that'd be clear.